indexing-service/src/main/java/org/apache/druid/indexing/input/DruidSegmentReader.java

```diff
@@ -270,6 +273,17 @@ public boolean hasNext()
           theEvent.put(metric, value);
         }
       }
+
+      /*
+       * Timestamp is added last because we expect that the time column will always be a date time object.
+       * If it is added earlier, it can be overwritten by metrics or dimenstions with the same name.
+       *
+       * If a user names a metric or dimension `__time` it will be overwritten. This case should be rare since
```

Comment:
Hmm, I think this overwriting should never happen, but it could happen for some reason in practice, e.g., user mistake. How about logging a warning if there are duplicate column names? Doc could say some kind of warning messages can be printed if there are duplicates.

Reply:
I'm worried about log explosion. since this is done per row. I'd have to add explicit checking outside of this next block. Maybe in the constructor? Would that be visible enough in the logs?

Reply:
Ah good point. Now I think we need some schema validation for ingestion which could probably be done in `DataSchema`. But this would be a larger issue than the bug this PR fixes, and I'm ok with adding it later.